tests/lib/hardware/libvirt.py

```diff
@@ -140,6 +153,25 @@ def _backing_file_create(self):
         logger.info(f"node {self.name}: created qcow2 backing file under"
                     f"{self._snap_img_path}")
 
+    def _data_disk_count(self):
+        domdict = xmltodict.parse(self._dom.XMLDesc())
+        # We have two device by default - OS and Cloudinit seed
+        return len(domdict['domain']['devices']['disk']) - 2
```

Comment:
This feels like a hack. Could we use libvirt's storage pools to collect the added disks and then interate over the pool?

Reply:
I did it this way because pool were not used in the initial VM creation so this would add some complexity.
Given that the code is handling a "test" and not a long running VM, I'm not sure using a pool is important.
If we do add a pool, then it would make sense to add it for boot device too, I'm just not sure of the added value.

Reply:
Updated to use XML after discussion with Thomas.